flake.nix

```diff
@@ -0,0 +1,58 @@
+{
+  description = "Codex build flake";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+
+        # Create a dummy lsb_release script
+        dummy-lsb-release = pkgs.writeScriptBin "lsb_release" ''
+          #!${pkgs.stdenv.shell}
+          echo "Distributor ID: NixOS"
+          echo "Description:    NixOS"
+          echo "Release:        Unstable"
+          echo "Codename:       nixos"
+        '';
+      in {
+        devShell = pkgs.mkShell {
+            buildInputs = with pkgs; [
+                gcc11
```

Comment:
I think the more kosher way of doing this would involve changing the GCC for the whole shell:
```nix
mkShell = mkShell.override { stdenv = pkgs.gcc11Stdenv; };
```
Something like that.